src/Command/CommandBase.php

```diff
@@ -2483,4 +2489,92 @@ protected function hasExternalGitHost(Project $project)
 
         return $ssh->hostIsInternal($project->getGitUrl()) === false;
     }
+
+    /**
+     * Loads the next environment deployment and caches it statically.
```

Comment:
Can you move these new methods to the `Api.php` service rather than CommandBase?